presto-main/src/test/java/com/facebook/presto/sql/planner/TestDomainTranslator.java

```diff
@@ -1321,6 +1324,48 @@ public void testDecimalComparedToWiderDecimal()
                 Range.lessThan(createDecimalType(4, 2), shortDecimal("99.99")), Range.greaterThan(createDecimalType(4, 2), shortDecimal("99.99"))), true));
     }
 
+    @Test
+    public void testVarcharComparedToCharExpression()
+            throws Exception
+    {
+        // greater than or equal
+        testSimpleComparison(greaterThanOrEqual(P, stringLiteral("123456789")), P, Range.greaterThan(createCharType(10), Slices.utf8Slice("12345678")));
```

Comment:
Please use `"abcd..."` instead of `"12345...."` for chars.
This is somewhat disturbing.